### ISSUE TITLE
Strange stop gradient operation

### ISSUE THREAD

**Issue by yoctta** [adobe-research/custom-diffusion#42]
What is the purpose of these codes in diffuser_training.py?
        if crossattn:
            modifier = torch.ones_like(key)
            # print(key.shape)
            modifier[:, :1, :] = modifier[:, :1, :]*0.
            key = modifier*key + (1-modifier)*key.detach()
            value = modifier*value + (1-modifier)*value.detach()

It looks like only use the first token to train to_k and to_v and not allowing gradient flow to the modifier token.

**Comment by duxiaodan** [comment]
@yoctta I'm also confused by this part. I found same code here https://github.com/adobe-research/custom-diffusion/blob/040f5fdaae64e8cb400fca891a0a3256244a9180/src/model.py#L154-L158. Maybe it's copy-and-pasted from the ldm codebase and is not necessary here? @nupurkmr9 could you please confirm my guess?

**Comment by nupurkmr9** [comment]
Hi, we stop the gradient on the first token of the text transformer output since its always the same and I observed small benefits when doing that.

**Comment by duxiaodan** [comment]
Oh I see. You are right. Thank you very much for the clarification. Custom Diffusion is a great work!
